Xamarin.Forms.Platform.iOS/Renderers/ListViewRenderer.cs

```diff
@@ -984,6 +1006,29 @@ void UpdateShortNameListener()
 						((INotifyCollectionChanged)templatedList.ShortNames).CollectionChanged -= OnShortNamesCollectionChanged;
 				}
 			}
+
+			protected override void Dispose(bool disposing)
+			{
+				if (_disposed)
+					return;
+
+				if (disposing)
+				{
+					if (List != null)
+					{
+						List.ItemSelected -= OnItemSelected;
+						List = null;
+					}
+
+					_templateToId = null;
+					_uiTableView = null;
+					_uiTableViewController = null;
+				}
+
+				_disposed = true;
+
+				base.Dispose(disposing);
+			}
```

Comment:
I did not see `OnItemSelected` being unregistered anywhere. References to `ListView` and `FormsUITableViewController` must be nullified. I think what is happening is when the constructor is called, we are passing the references by value which creates a copy for each. You could test this particular change by having a list view with no items. If these references are not nullified, you end up with a leak still.

I'm not sure if setting `_templateToId` to null was required since it's a `Dictionary`, but the keys are data templates. It seemed safer to null it out as well.